pkg/sql/ttl/ttljob/ttljob.go

```diff
@@ -723,7 +730,14 @@ func runTTLOnRange(
 			break
 		}
 	}
-	return nil
+	return db.Txn(ctx, func(_ context.Context, txn *kv.Txn) error {
+		return job.Update(ctx, txn, func(_ *kv.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
```

Comment:
Is `job.Update` safe to call concurrently?

Reply:
will this be racy tho if two ranges try updates the row count at the same time?

Reply:
we potentially also want this in a `defer` in case the deletes fail for some other reason.

Reply:
>  I can't tell what causes it to take other concurrent updates into account though when it is called.

the SELECT FOR UPDATE should help:

https://github.com/cockroachdb/cockroach/blob/b8eee78e455ad97f85f6a74c22732d3c5f3ee349/pkg/jobs/update.go#L319

but that's currently false:
https://github.com/cockroachdb/cockroach/blob/b8eee78e455ad97f85f6a74c22732d3c5f3ee349/pkg/jobs/update.go#L138

so not quite sure it's concurrency safe - @miretskiy can you elaborate?

> Is there a way to cause a partial failure for a test?

you would need to mock the database calls and insert that in :D. actually, it's probably easiest to use the `sql.TTLTestingKnobs` to insert some bool/function hook to error when executing a query (look at how other testing knobs are used)